src/cow-react/modules/trade/hooks/setupTradeState/useSetupTradeState.ts

```diff
@@ -53,13 +53,16 @@ function areChainIdsTheSame(aChainId: Nullish<number>, bChainId: Nullish<number>
 }
 
 export function useSetupTradeState(): void {
-  const { chainId: currentChainId, connector, account } = useWeb3React()
+  const { chainId: providerChainId, connector, account } = useWeb3React()
   const [isChainIdSet, setIsChainIdSet] = useState(false)
   const tradeNavigate = useTradeNavigate()
   const tradeStateFromUrl = useTradeStateFromUrl()
   const tradeState = useTradeState()
 
   const chainIdFromUrl = tradeStateFromUrl.chainId
+  // When there is no account from provider, then we consider provider as not connected and use chainId from URL as current
+  const currentChainId = (account ? providerChainId : chainIdFromUrl) || providerChainId
```

Comment:
I suggest to highlight this change is not purely e2e tests, but also a behaviour change/bug fix

Reply:
Thanks! Added description